scripts/foundry/contract-upgrades/v3.1.0/README.md

```diff
@@ -0,0 +1,121 @@
+...
+Done.
+```
+
+9. Run the upgrade script. If the `L1_PRIV_KEY` is not the chain owner, it will print the upgrade payload. Execute the payload with the chain owner.
+
+> [!CAUTION]
+> This script will not ask for confirmation before sending the transaction!
+
+```
+$ L1_PRIV_KEY=xxx yarn script:bold-local-execute --network {mainnet|arb1|base|arbSepolia}
+upgrade executor: 0x5FEe78FE9AD96c1d8557C6D6BB22Eb5A61eeD315
+execute(...) call to upgrade executor: 0x1cff79cd000000000000000000000000f8199ca3702c09c78b957d4d820311125753c6d2000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000a4ebe03a93000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000030000000000000000000000008a8f0a24d7e58a76fc8f77bb68c7c902b91e182e00000000000000000000000087630025e63a30ecf9ca9d580d9d95922fea6af0000000000000000000000000c32b93e581db6ebc50c08ce381143a259b92f1ed00000000000000000000000000000000000000000000000000000000
+```
+
+10. Run the following command to update the `EspressoTEEVerifier` address in SequencerInbox contract. Note: The `PARENT_CHAIN_UPGRADE_EXECUTOR` is the address of the upgrade executor contract on the parent chain, the `SequencerInboxProxyAddress` is the address of the SequencerInbox proxy contract on the parent chain, and the `EspressoTEEVerifierAddress` is the address of the EspressoTEEVerifier contract on the child chain which is the same address you put in custom.ts.
+
+```
+cast send $PARENT_CHAIN_UPGRADE_EXECUTOR $(cast calldata "executeCall(address, bytes)" $SequencerInboxProxyAddress $(cast calldata "setEspressoTEEVerifier(address)" $EspressoTEEVerifierAddress))  --rpc-url $PARENT_CHAIN_RPC_URL --private-key $PRIVATE_KEY
+```
+
+11. Update your nodes configuration and restart them:
+
+- For validator nodes: add `--node.bold.enable=true` and `--node.bold.strategy=<MakeNodes | ResolveNodes | Defensive>` to configure the validator to create and/or confirm assertions in the new Rollup contract (find more information in [How to run a validator](https://docs.arbitrum.io/run-arbitrum-node/more-types/run-validator-node#step-1-configure-and-run-your-validator))
+- For all other types of node before v3.6.0: add `--node.bold.enable=true` to enable [watchtower mode](https://docs.arbitrum.io/run-arbitrum-node/run-full-node#watchtower-mode)
+- For all other types of node after Nitro v3.6.0: [watchtower mode](https://docs.arbitrum.io/run-arbitrum-node/run-full-node#watchtower-mode) is automatically enabled
+
+Additionally, after performing the upgrade, the `--chain.info-json` object also needs to be modified:
+
+- Update the new rollup address in the `rollup.rollup` field
+- Add the stake token in a new `rollup.stake-token` field
+
+12. If Espresso team is running the batch poster, wait for Espresso team to also run the batch poster with the updated config and then move to the next step.
```

Comment:
Isn't this something that we could be doing ahead of time? Like I think it could be a useful "Important info" addendum to number 11 but idk if I would call this a step in the process?

Reply:
Okay yeah I was reading this like they had to avoid starting the validator until we have the batcher updated, not that they just have to wait to monitor assertions until after we have updated the batcher.

I guess this is fine then. Sorry for the confusion!